compiler/rustc_middle/Cargo.toml

```diff
@@ -8,6 +8,7 @@ edition = "2018"
 doctest = false
 
 [dependencies]
+either = "1.5.0"
```

Comment:
Just a note, this is an extra dependency, don't know how willing we are to accept that.